sycl/include/CL/__spirv/spirv_types.hpp

```diff
@@ -110,9 +110,25 @@ enum class GroupOperation : uint32_t {
 
 enum class MatrixLayout { RowMajor, ColumnMajor, PackedA, PackedB };
 
+// TODO: replace the following W/A with a better solution when we have it.
+// The following structure is used to represent the joint matrix type in the
+// LLVM IR. The structure has a pointer to a multidimensional array member which
+// makes the encoding of the matrix type information within the LLVM IR looks
+// like this:
+// %struct.__spirv_JointMatrixINTEL = type { [42 x [6 x [2 x [1 x float]]]]* }
+// Note that an array cannot be of zero size but MatrixLayout and Scope
+// parameters can; hence '+ 1' is added to the 3rd and 4th dimensions.
```

Comment:
Are `MatrixLayout` and `Scope` unsigned integers? If not, the trick with adding one doesn't work.

Reply:
They are both enums with numbering started from '0'. Guess, may be:
`enum class MatrixLayout { RowMajor, ColumnMajor, PackedA, PackedB };`
can be changed to:
`enum class MatrixLayout : unsigned { RowMajor, ColumnMajor, PackedA, PackedB };`
to make it impossible to start it from '-1', but not sure if it worth it.

Reply:
I'll add explicit numbering in the next patch set.

Reply:
Totally worth it. Please, document all restrictions you put on these enums.
`enum class MatrixLayout : int { RowMajor = -1, ...};` seems to be a valid definition.

Reply:
Done, thanks!